main.py

```diff
@@ -1,129 +1,156 @@
-    INSERT INTO weather (bikeable, icon, lat, lng, saved_at, temperature)
-    VALUES (?, ?, ?, ?, ?, ?);
-    """
-    values = (
-        weather_data['bikeable'],
-        weather_data['icon'],
-        weather_data['location']['lat'],
-        weather_data['location']['lng'],
-        weather_data['saved_at'],
-        weather_data['temperature']
-    )
-    db_client.insert(keys=('bikeable', 'icon', 'lat', 'lng', 'saved_at', 'temperature'), values=values, table='weather')
 
 def get_current_location():
     try:
         location_url = f'http://ipinfo.io/?token={ipinfo_api_token}'
         response = requests.get(location_url)
 
-        if response.status_code == 200:
-            location_data = response.json()
-            coordinates = location_data.get('loc', '').split(',')
-            if len(coordinates) == 2:
-                latitude, longitude = map(float, coordinates)
-                return latitude, longitude
-            else:
-                return None, None
-        else:
+        if not response.status_code == 200:
             return None, None
+
+        location_data = response.json()
+        coordinates = location_data.get('loc', '').split(',')
+
+        if not len(coordinates) == 2:
+            return None, None
+
+        latitude, longitude = map(float, coordinates)
+        return latitude, longitude
     except Exception as e:
+        print(f"Server not found! Error:{e}")
         return None, None
 
+
 def get_weather(latitude, longitude):
-    api_key = os.getenv('WEATHER_API_KEY')
-    api_url_base = os.getenv('WEATHER_API_URL', 'https://dataservice.accuweather.com')
     api_url = f'{api_url_base}/locations/v1/cities/geoposition/search'
-
     params = {
         'apikey': api_key,
         'q': f'{latitude},{longitude}'
     }
 
     response = requests.get(api_url, params=params)
 
-    if response.status_code == 200:
-        location_data_unformatted = response.json()
-        location_data = [location_data_unformatted]
-        location_key = location_data[0]['Key']
-        api_url = f'{api_url_base}/currentconditions/v1/{location_key}'
+    if not response.status_code == 200:
+        print("There seems to be an issue")
+        # Handle the error appropriately
 
-        params = {
-            'apikey': api_key,
-        }
+    location_data = [response.json()]
+    location_key = location_data[0]['Key']
+    api_url = f'{api_url_base}/currentconditions/v1/{location_key}'
 
-        weather_response = requests.get(api_url, params=params)
-        return weather_response
+    return requests.get(api_url, params=params)
 
-    else:
-        print("Er gaat duidelijk iets NIET goed")
+def parse_datetime(saved_at_str):
+    try:
+        return datetime.strptime(saved_at_str, '%Y-%m-%d %H:%M:%S')
+    except ValueError:
+        return None
 
 @app.route('/')
 def weather():
     create_weather_table()
     latitude, longitude = get_current_location()
 
+    # Check if data is in the cache
```

Comment:
You need to fetch the stored data from the database instead of using a {} what if the service restarts.

Reply:
Done

Reply:
Remove the comments if is finished

Reply:
all done